core/src/main/java/io/smallrye/context/SmallRyeManagedExecutor.java

```diff
@@ -386,10 +386,35 @@ public Builder injectionPointName(String name) {
             return this;
         }
 
+        /**
+         * Delegate execution to the given executor service instead of the default set by
+         * {@link SmallRyeContextManager.Builder#withDefaultExecutorService(ExecutorService)}. Set to <code>null</code> to
+         * disable delegation
+         * and force creating a new executor service.
+         * 
+         * @param executorService the executor service to delegate to, or <code>null</code> to force creating a new executor
+         *        service.
+         * @return this builder
+         * @see #withNewExecutorService()
+         * @see SmallRyeContextManager.Builder#withDefaultExecutorService(ExecutorService)
+         */
         public Builder withExecutorService(ExecutorService executorService) {
             this.executorService = executorService;
             return this;
         }
 
+        /**
+         * Forces the creation of a new executor service. This is has the same effect as calling
+         * {@link #withExecutorService(ExecutorService)} with
+         * <code>null</code>.
+         * 
+         * @return this builder
+         * @see #withExecutorService(ExecutorService)
+         * @see SmallRyeContextManager.Builder#withDefaultExecutorService(ExecutorService)
+         */
+        public Builder withNewExecutorService() {
```

Comment:
Do we really need this method since you can do the same with `withExecutorService()` and `null` value? Wouldn't it make more sense to either forbid `null` in `withExecutorService()` or remove this method?

Reply:
So, I don't feel strongly about this, but it seems to me that this name is more intuitive than passing `null` and so I think it may be useful for users to discover this API using this name.
If you feel strongly about this, I can remove this method.

Reply:
OK great, thanks :)